packages/engine/src/common/classes/GenerateMeshBVHWorker.ts

```diff
@@ -62,33 +62,30 @@ export class GenerateMeshBVHWorker {
         }
       }
 
-      const posAttribute = geometry.getAttribute('position')
-      const position = Float32Array.from(posAttribute.array)
-      if (!geometry.getIndex()) {
-        geometry.setIndex(Array.from(Object.keys(posAttribute.array).map(Number)))
-      }
-      const index = Float32Array.from(geometry.getIndex().array)
-
       if (
-        (posAttribute as InterleavedBufferAttribute).isInterleavedBufferAttribute ||
-        (geometry.getIndex() as any as InterleavedBufferAttribute)?.isInterleavedBufferAttribute
+        (geometry.attributes.position as InterleavedBufferAttribute).isInterleavedBufferAttribute ||
+        (geometry.index && (geometry.index as any as InterleavedBufferAttribute).isInterleavedBufferAttribute)
       ) {
-        console.warn(
-          'GenerateMeshBVHWorker: InterleavedBufferAttribute are not supported for the geometry attributes.',
-          geometry
+        throw new Error(
+          'GenerateMeshBVHWorker: InterleavedBufferAttribute are not supported for the geometry attributes.'
         )
-        return
       }
 
-      const transferrables = [position.buffer, index.buffer]
+      const index = geometry.index ? geometry.index.array : null
```

Comment:
transferring the original copy and having this async process will lead to the geometry data being undefined for a short period of time, which will cause errors in other areas.  This is the reason i did a copy of them.

Reply:
Yes, I'm aware of this problem. But the previous implementation had two issues:

- The raycasting did not work after the BVH generation is finished
- The buffer copy operation made generation slower and took more memory because GC does not kick in instantly after reference invalidation and when it does, it introduces jitter.